crates/ide_completion/src/render/function.rs

```diff
@@ -228,7 +228,7 @@ fn should_add_parens(ctx: &CompletionContext) -> bool {
 }
 
 fn detail(db: &dyn HirDatabase, func: hir::Function) -> String {
-    let ret_ty = func.ret_type(db);
+    let ret_ty = func.async_ret_type(db).unwrap_or_else(|| func.ret_type(db));
```

Comment:
Nit: I think this would be clearer with the `is_async` check pulled outside, but it doesn't matter too much.

Reply:
Done! I also left that check inside `hir::Function::async_ret_type` in case someone misuse it on `impl AnyOtherTrait` types